Polyhedron/demo/Polyhedron/Scene_surface_mesh_item.cpp

```diff
@@ -1,30 +1,68 @@
 #include "Scene_surface_mesh_item.h"
 
+#include "Color_map.h"
 #include <queue>
 #include <boost/graph/properties.hpp>
 #include <boost/graph/graph_traits.hpp>
 #include <QOpenGLShaderProgram>
 #include <QOpenGLBuffer>
 #include <QApplication>
+#include <QVariant>
 
-#include <CGAL/boost/graph/properties_Surface_mesh.h>
-#include <CGAL/Surface_mesh/Surface_mesh.h>
+//#include <CGAL/boost/graph/properties_Surface_mesh.h>
+#include <CGAL/Surface_mesh.h>
+#include <CGAL/intersections.h>
+#include <CGAL/AABB_tree.h>
+#include <CGAL/AABB_traits.h>
 
 #include <CGAL/Polygon_mesh_processing/compute_normal.h>
 #include "triangulate_primitive.h"
 
 typedef boost::graph_traits<Scene_surface_mesh_item::SMesh>::face_descriptor face_descriptor;
 typedef boost::graph_traits<Scene_surface_mesh_item::SMesh>::halfedge_descriptor halfedge_descriptor;
 typedef boost::graph_traits<Scene_surface_mesh_item::SMesh>::vertex_descriptor vertex_descriptor;
+//Used to triangulate the AABB_Tree
+class Primitive
+{
+public:
+  // types
+  typedef face_descriptor Id; // Id type
+  typedef Scene_surface_mesh_item::Point Point; // point type
+  typedef Scene_surface_mesh_item::Kernel::Triangle_3 Datum; // datum type
+
+private:
+  // member data
+  Id m_it; // iterator
+  Datum m_datum; // 3D triangle
+
+  // constructor
+public:
+  Primitive() {}
+  Primitive(Datum triangle, Id it)
+    : m_it(it), m_datum(triangle)
+  {
+  }
+public:
+  Id& id() { return m_it; }
+  const Id& id() const { return m_it; }
+  Datum& datum() { return m_datum; }
+  const Datum& datum() const { return m_datum; }
+
+  /// Returns a point on the primitive
+  Point reference_point() const { return m_datum.vertex(0); }
+};
```

Comment:
after discussing with @maxGimeno it turns out that its a special primitive to handle non-triangulated meshes so a dedicated one is needed. It can probably be shared with polyhedron.